repo/README.md

```diff
@@ -86,26 +74,53 @@ feature swappable datastores, for example:
 This makes it easy to change properties or performance characteristics of
 a repo without an entirely new implementation.
 
+
+### keys (state)
+
+A Repo typically holds the keys a node has access to, for signing and for encryption. This includes:
+
+- a special (private, public) key pair that defines the node's identity
+- (private, public) key pairs
+- symmetric keys
+
+Some repos MAY support key-agent delegation, instead of storing the keys directly.
+
+Keys are structured using the [multikey](https://github.com/jbenet/multikey) format, and are part of the [keychain](../keychain) datastructure. This means all keys are IPLD objects, and that they link to all the data needed to make sense of them, including parent keys, identities, and certificates.
+
+### config (state)
+
+The node's `config` (configuration) is a tree of variables, used to configure various aspects of operation. For example:
+- the set of bootstrap peers IPFS uses to connect to the network
+- the Swarm, API, and Gateway network listen addresses
+
+It is recommended that `config` files avoid identifying information, so that they may be re-shared across multiple nodes.
+
+**CHANGES**: today, implementations like go-ipfs store the peer-id and private key directly in the config. These will be removed and moved out.
```

Comment:
Would be good to specify the json schema? Ideally any implementation should be compatible.